components/launcher-client/src/client.rs

```diff
@@ -36,7 +36,7 @@ pub struct LauncherCli {
 
 impl Drop for LauncherCli {
     fn drop(&mut self) {
-        if fs::remove_file(&self.pipe).is_err() {
+        if cfg!(unix) && fs::remove_file(&self.pipe).is_err() {
```

Comment:
If Windows doesn't need to do anything here, why not make the whole `impl Drop` `#[cfg(not(windows))]`? As is, it will misleadingly print "Removed old pipe to launcher" on Windows, when this code has done no such thing.

Also, I suggest `(not(windows))` that rather than `(unix)` since I think we should err on the side of doing the cleanup, and experience will tell us when the cleanup we're doing is wrong. If we err on the side of doing nothing, we won't get any signals that we should be doing something.